core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
```

Comment:
So, if training data changes, it goes through an exploration commit. When that happens, a comparison is pulled up between the old job and the new changes. So, if the job isn't picked up by the VM yet, it ideally, we can just invalidate the older job and let the new one be picked up. I don't think we need to let the controller populate training data because of that.